### PR TITLE
XML Documentation Fix - Batch #2

### PR DESCRIPTION
**Pull request by benrr101**
Description: This PR is the first of several that aim to fix https://github.com/dotnet/SqlClient/issues/2013. As per discussion on that issue, this PR removes Markdown style comments and replaces them with XML style comments.

This batch also takes into consideration the comments from the previous PR, so hopefully there will be less corrections required on this PR.

One open question - the `cref` format requires using a prefix for the type of the thing being reference (eg, `M:` for method, `T:` for types). However, it's unclear how the reference should be done for overloads, either specifying all overloads or specifying a single overload. The sandcastle docs say `O:` can be used to refer to all overloads (https://ewsoftware.github.io/XMLCommentsGuide/html/983fed56-321c-4daf-af16-e3169b28ffcd.htm#Examples)? The markdown docs use `%2A` (`*`) in cref's quite a bit, which I think effectively refers to all overloads, and those should be replaced with `O:` prefix? Please let me know any thoughts on this

### CODE REVIEW

File: doc/snippets/Microsoft.Data.SqlClient/SqlBatchCommandCollection.xml [hunk 1]
```diff
@@ -146,31 +135,46 @@
       <param name="index">
         The index where the <see cref="T:Microsoft.Data.SqlClient.SqlBatchCommand" /> object should be located.
       </param>
-      <param name="batchCommand">To be added.</param>
       <summary>
```
Comment (David-Engel) [inline doc/snippets/Microsoft.Data.SqlClient/SqlBatchCommandCollection.xml:138]: Build break here:
```suggestion
      <param name="batchCommand">
          The <see cref="T:Microsoft.Data.SqlClient.SqlBatchCommand" /> object.
      </param>
      <summary>
```

Comment (benrr101) [reply doc/snippets/Microsoft.Data.SqlClient/SqlBatchCommandCollection.xml:141]: Thanks for that - I've fixed in the most recent commit.

**Review by JRahnama — APPROVED** [review]

**Comment by JRahnama** [comment]
on the built M.D.SqlClient.xml in artifact folder I see
```xml
 <member name="M:Microsoft.Data.SqlClient.SqlBulkCopyColumnMapping.#ctor(System.String,System.String)">
            <!-- No matching elements were found for the following include tag --><include file="../../../../../../doc/snippets/Microsoft.Data.SqlClient/SqlBulkCopyColumnMapping.xml" path="docs/members[@name=&quot;SqlBulkCopyColumnMapping&quot;]/ctor[@name=&quot;sourceColumnStringAnddestinationColumnString&quot;]/*" />
        </member>
```

**Comment by benrr101** [comment]
@JRahnama thanks for pointing that out. I'll add the fix for it to the next PR.

**Comment by benrr101** [comment]
@JRahnama sorry I though I already merged this one. I'll cherry pick the fixes from the next pr back into this one!

**Pull request merged**